CMakeLists.txt

```diff
@@ -788,8 +788,8 @@ IF(XNNPACK_BUILD_LIBRARY)
   TARGET_LINK_LIBRARIES(operator-run PRIVATE xnnpack-base logging)
   TARGET_LINK_LIBRARIES(operator-utils PRIVATE xnnpack-base logging)
   TARGET_LINK_LIBRARIES(subgraph PRIVATE xnnpack-base allocator logging memory mutex operators operator-run)
-  TARGET_LINK_LIBRARIES(XNNPACK PRIVATE xnnpack-base allocator cache hardware-config indirection logging memory microkernel-utils microparams-init mutex normalization operators operator-run operator-utils packing microkernels-prod subgraph)
-  TARGET_LINK_LIBRARIES(XNNPACK PUBLIC pthreadpool)
+  TARGET_LINK_LIBRARIES(XNNPACK PRIVATE allocator cache hardware-config indirection memory microkernel-utils microparams-init mutex normalization operators operator-run operator-utils packing microkernels-prod subgraph)
+  TARGET_LINK_LIBRARIES(XNNPACK PUBLIC xnnpack-base logging)
```

Comment:
Does this also work if you do
```
TARGET_LINK_LIBRARIES(XNNPACK PUBLIC pthreadpool logging)
```
i.e. leave `xnnpack-base` in the `PRIVATE` linkeage statement above?

Asking because having `xnnpack-base` here was causing problems (and was recently removed in #7056) since it propagates all the associated compiler flags with it.

Reply:
Yes, that worked for me as well! I’ve made a new commit to reflect this fix. Thank you for the suggestion!